src/Symfony/Component/Security/Csrf/TokenStorage/CookieTokenStorageListener.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Csrf\TokenStorage;
+
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+
+/**
+ * Checks the request's attributes for a CookieTokenStorage instance. If one is
+ * found, the cookies representing the storage's changeset are appended to the
+ * response headers.
+ *
+ * TODO where to put this class?
+ *
+ * @author Oliver Hoff <oliver@hofff.com>
+ */
+class CookieTokenStorageListener
+{
+    /**
+     * @var string
+     */
+    private $tokenStorageKey;
+
+    /**
+     * @param string|null $tokenStorageKey
+     */
+    public function __construct($tokenStorageKey = null)
+    {
+        // TODO should this class get its own DEFAULT_TOKEN_STORAGE_KEY constant?
+        // if no, where should the sole constant be put?
+        $this->tokenStorageKey = $tokenStorageKey === null ? RequestStackTokenStorage::DEFAULT_TOKEN_STORAGE_KEY : $tokenStorageKey;
```

Comment:
well its also not specific to TokenStorageInterface, but to code managing TokenStorageInterfaces within higher aggregates. the managing code doesnt have to be a TokenStorageInterface itself (like the RequestStackTokenStorage), but can be anything (like a listener in the case of CookieTokenStorageListener). its also just a default key, if no explicit one is passed, to the managing code. each class could provide its own default key, but it would be handy, if every implementation uses the same default, especially when they are meant to be used together (like the RequestStackTokenStorage and the CookieTokenStorageListener).